client/my-sites/upgrades/trials/data/free-trial-cart.jsx

```diff
@@ -0,0 +1,93 @@
+/**
+ * External dependencies
+ */
+import React from 'react';
+
+/**
+ * Internal dependencies
+ */
+import cartValues, { cartItems } from 'lib/cart-values';
+import observe from 'lib/mixins/data-observe';
+import TransactionStore from 'lib/transaction/store';
+import StoreConnection from 'components/data/store-connection';
+import { resetTransaction, setPayment } from 'lib/upgrades/actions';
+import storeTransactions from 'lib/store-transactions';
+
+const FreeTrialCartData = React.createClass( {
+	mixins: [ observe( 'sites', 'products' ) ],
+
+	getInitialState: function() {
+		return { cart: {} };
+	},
+
+	componentWillMount: function() {
+		// reset the transaction store
+		resetTransaction();
+
+		// FIXME: The endpoint doesn't currently support transactions with no
+		//   payment info, so for now we rely on the credits payment method for
+		//   free carts.
+		setPayment( storeTransactions.fullCreditsPayment() );
+
+		this.updateCart();
+	},
+
+	componentDidMount: function() {
+		this.props.sites.on( 'change', this.updateCart );
+	},
+
+	componentWillUnmount: function() {
+		this.props.sites.off( 'change', this.updateCart );
+	},
+
+	updateCart: function() {
+		if ( this.isLoading() ) {
+			return;
+		}
+
+		const planSlug = this.props.plans.getSlugFromPath( this.props.planName ),
+			planItem = cartItems.getItemForPlan( { product_slug: planSlug }, { isFreeTrial: true } );
+
+		this.replaceCartWithItem( planItem );
+	},
+
+	replaceCartWithItem: function( item ) {
+		const selectedSite = this.props.sites.getSelectedSite(),
+			emptyCart = cartValues.emptyTemporaryCart( selectedSite.ID );
+
+		const newCart = cartValues.fillInAllCartItemAttributes(
+			cartItems.add( item )( emptyCart ),
+			this.props.products.get()
+		);
+
+		this.setState( { cart: newCart } );
+	},
+
+	isLoading: function() {
```

Comment:
We've been naming these methods `isDataLoading()` in other components so maybe we should follow this pattern here.